File: Data Structures/BinaryTrees/ChildrenSum.py
class Node:

    def __init__(self, data):
        self.data = data
        self.left = self.right = None

def checkChildrenSum(node):
    
    left_data = 0
    right_data = 0

    if node is None or (node.left is None and node.right is None):
        return True
    else:
        if node.left is not None:
            left_data = node.left.data
        if node.right is not None:
            right_data = node.right.data
        if node.data == right_data + left_data and checkChildrenSum(node.left) and checkChildrenSum(node.right):
            return True
        else:
            return False

File: Data Structures/BinaryTrees/ChildrenSum.java
class ChildrenSum
{
    static class Node
    {
        int data;
        Node left, right;
        public Node(int data)
        {
            this.data = data;
            this.left = this.right = null;
        }
    }

    static boolean checkChildrenSum(Node root)
    {
        int left_data = 0, right_data = 0;
        if(root == null || (root.left == null && root.right == null))
            return true;
        else
        {
            if(root.left != null)
                left_data = root.left.data;
            if(root.right != null)
                right_data = root.right.data;
            if( (root.data == left_data + right_data) && (checkChildrenSum(root.left)) && checkChildrenSum(root.right) )
                return true;
            else
                return false;
        }
    }
}


